src/main/scala/uncore/tilelink2/ToAXI4.scala

```diff
@@ -199,17 +199,18 @@ class TLToAXI4(beatBytes: Int, combinational: Boolean = true, adapterName: Optio
       val a_sel = UIntToOH(arw.id, edgeOut.master.endId).toBools
       val d_sel = UIntToOH(Mux(r_wins, out.r.bits.id, out.b.bits.id), edgeOut.master.endId).toBools
       val d_last = Mux(r_wins, out.r.bits.last, Bool(true))
-      (a_sel zip d_sel zip idStall zip idCount) filter { case (_, n) => n > 1 } foreach { case (((as, ds), s), n) =>
-        val count = RegInit(UInt(0, width = log2Ceil(n + 1)))
+      // If FIFO was requested, ensure that R+W ordering is preserved
+      (a_sel zip d_sel zip idStall zip idCount) filter { case (_, n) => n.map(_ > 1).getOrElse(false) } foreach { case (((as, ds), s), n) =>
```

Comment:
`(a_sel zip d_sel zip idStall zip idCount) foreach { case (((as, ds), s), n) if n.map(_ > 1).getOrElse(false) =>` for added cuteness?